File: javascript/1200. Minimum Absolute Difference.js
/**
 * @param {number[]} arr
 * @return {number[][]}
 */
var minimumAbsDifference = function(arr) {
    arr.sort((a,b) => a-b);
    let minDiff = Infinity;
    let pairs = [];

    let set = new Set(arr);
    for(let index = 0; index < arr.length-1; index++){
        minDiff = Math.min(minDiff, Math.abs(arr[index+1]-arr[index]));
    }
    for(let index = 0; index < arr.length; index++){
        let number = arr[index];
        let diff = number + minDiff;
        if(set.has(diff) == true){
            pairs.push([number, diff]);
            continue;
        }
    }
    return pairs;
};

File: java/1200. Minimum Absolute Difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        Arrays.sort(arr);
        List<List<Integer>> pairs = new LinkedList();
        int minDiff = Integer.MAX_VALUE;

        
        
        for(int index = 1; index < arr.length; index++){
            minDiff = Math.min(minDiff, arr[index]-arr[index-1]);
        }
        
        for(int index = 1; index < arr.length; index++){
            int diff = arr[index] - arr[index-1];
            if(diff == minDiff){
                List pair = new LinkedList();
                pair.add(arr[index-1]);
                pair.add(arr[index]);
                pairs.add(pair);
                
            }
        }
        return pairs;
    }
}
